controllers/healthcheck_controller.go

```diff
@@ -290,3 +343,148 @@ func (r *HealthCheckReconciler) parseWorkflowFromHealthcheck(log logr.Logger, hc
 	uwf.SetUnstructuredContent(content)
 	return nil
 }
+
+// Create ServiceAccount
+func (r *HealthCheckReconciler) CreateServiceAccount(clientset kubernetes.Interface, name string, namespace string) (string, error) {
+	sa, err := clientset.CoreV1().ServiceAccounts(namespace).Get(name, metav1.GetOptions{})
+	// If a service account already exists just re-use it
+	if err == nil {
+		return sa.Name, nil
+	}
+
+	sa = &v1.ServiceAccount{
```

Comment:
I like the nice and straight-forward pattern used in these methods

still a bit of repeated code but that could always be trimmed out in following change